### PR TITLE
PODS-9965: Optimising the json reading

### PR DESCRIPTION
**Pull request by joecefnhedog**

### CODE REVIEW

File: app/controllers/AFTController.scala [hunk 1]
```diff
@@ -242,15 +242,14 @@ class AFTController @Inject()(
       }
   }
 
-  private def detailsJsLogic(js:JsValue,version:AFTVersion): VersionsWithSubmitter = {
+
+
+  private def detailsJsLogic(js: JsValue, version: AFTVersion): VersionsWithSubmitter = {
+    import models.AFTSubmitterDetails._
     logger.warn(s"detailsJsLogic started for: ${version.reportVersion}")
-    val transform = js.transform(aftDetailsTransformer.transformToUserAnswers) match {
```
Comment (joecefnhedog) [inline app/controllers/AFTController.scala:247]: We were reading all the data to userAnswers first, then when we get user answers we read it again to get AFTSubmitterDetails. 

I have changed it to read straight into AFTSubmitterDetails.

**Comment by joecefnhedog** [comment]
I mocked up a big json payload:
image
Performed the request before and after:

old version ~ 1 second
2024-11-28 00:39:58,669 start
2024-11-28 00:39:59,609 end

new version ~ 1 microsecond?:
2024-11-28 00:41:55,557
2024-11-28 00:41:55,558

obviously needs testing multiple times, but it seems promising.

full log:
`2024-11-28 00:39:58,669 level=[WARN] logger=[controllers.AFTController] thread=[application-pekko.actor.default-dispatcher-7] rid=[govuk-tax-f11cab4f-b566-496f-9433-453ee05dd047] user=[] message=[detailsJsLogic started for: 1]
old version
2024-11-28 00:39:59,609 level=[WARN] logger=[controllers.AFTController] thread=[application-pekko.actor.default-dispatcher-7] rid=[govuk-tax-f11cab4f-b566-496f-9433-453ee05dd047] user=[] message=[detailsJsLogic finished for: 1]

2024-11-28 00:41:55,557 level=[WARN] logger=[controllers.AFTController] thread=[application-pekko.actor.default-dispatcher-7] rid=[govuk-tax-a5150ef9-637f-4a5c-bc59-88222665bc20] user=[] message=[detailsJsLogic started for: 1]
new version
2024-11-28 00:41:55,558 level=[WARN] logger=[controllers.AFTController] thread=[application-pekko.actor.default-dispatcher-7] rid=[govuk-tax-a5150ef9-637f-4a5c-bc59-88222665bc20] user=[] message=[detailsJsLogic finished for: 1]`

**Review by dynamicwave4 — APPROVED** [review]

File: app/models/AFTSubmitterDetails.scala [hunk 2]
```diff
@@ -16,14 +16,27 @@
...
   implicit val formats: Format[AFTSubmitterDetails] = Json.format[AFTSubmitterDetails]
+
+  implicit val readAftDetailsFromIF: Reads[AFTSubmitterDetails] = Reads { json =>
+    for {
+      submitterType <- (json \ "aftDeclarationDetails" \ "submittedBy").validate[String]
+      submitterName <- (json \ "aftDeclarationDetails" \ "submitterName").validate[String]
+      submitterID <- (json \ "aftDeclarationDetails" \ "submitterId").validate[String]
+      authorisingPsaId = (json \ "aftDeclarationDetails" \ "psaId").asOpt[String]
+      receiptDate <- (json \ "aftDetails" \ "receiptDate").validate[String].map { dateTime =>
+        LocalDateTime.parse(dateTime.dropRight(1)).toLocalDate
```
Comment (joecefnhedog) [inline app/models/AFTSubmitterDetails.scala:35]: This is copying how they did it: https://github.com/hmrc/pension-scheme-accounting-for-tax/blob/main/app/transformations/ETMPToUserAnswers/AFTDetailsTransformer.scala#L82-L89

**Review by PavelVjalicin — APPROVED** [review]
Looks good to me.

**Pull request merged**